routes/data/POST-blob.js

```diff
@@ -24,30 +24,44 @@ module.exports = (app) => {
 
         // get the uploads directory path
         const uploadsDir = path.join(__dirname, '..', '..', 'uploads');
-        // creates the directory if it does not exist
-        if (!fs.existsSync(uploadsDir)) {
-          fs.mkdirSync(uploadsDir);
-        }
+        try {
+          // creates the directory if it does not exist
+          if (!fs.existsSync(uploadsDir)) {
+            fs.mkdirSync(uploadsDir);
+          }
 
-        // ensures the storage directory for the device's id exists
-        const storageDir = path.join(__dirname, '..', '..', 'uploads', deviceId);
-        if (!fs.existsSync(storageDir)) {
-          fs.mkdirSync(storageDir);
+          // ensures the storage directory for the device's id exists
+          const storageDir = path.join(__dirname, '..', '..', 'uploads', deviceId);
+          if (!fs.existsSync(storageDir)) {
+            fs.mkdirSync(storageDir);
+          }
+        } catch (err) {
+          next(err);
+          return res.status(500).json({ error: 'Server error while creating upload directory' });
         }
+      } else {
+        return res.status(400).json({ error: 'Only provide the device mac address under fieldname "deviceId"' });
       }
     }).on('fileBegin', (name, file) => {
       // this is called when a file upload is begining
       // specify the location that the file is uploaded to
+      if (deviceId === '' || deviceId == null) {
```

Comment:
Hi @Golem8 ,
Thank you for providing the updates. Let us ensure that we are sending the correct error-codes. As per the request, if a field is missing a 400 Bad Request should be sent. Please refer to [this ](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status)document for reference. This small fix should be made to ensure we are developing a user-friendly API.

Thank you,
-H